scripts/qa_crowbarsetup.sh

```diff
@@ -1609,18 +1609,59 @@ function set_proposalvars()
     # Determine if we went through an upgrade
     [ -f /etc/cloudsource ] && export cloudsource=$(</etc/cloudsource)
 
-    wantswift=1
-    [ -z "$want_swift" ] && wantceph=1
-    [[ -n "$wanthyperv" ]] && wantswift= && wantceph= && networkingmode=vlan
+    ### dynamic defaults
+    case "$nodenumber" in
+        0|1)
+            wantswift=
+            wantceph=
+        ;;
+        2)
+            wantswift=1
+            wantceph=
+        ;;
+        *)
+            wantswift=
+            wantceph=1
+        ;;
+    esac
+
+    ### filter (temporarily changing defaults)
+    # F1: hyperV only without swift and ceph
+    [ -n "$wanthyperv" ] && wantswift= && wantceph= && networkingmode=vlan
+
+    ### user requests (can override defaults and filters)
```

Comment:
Option b) would increase the amount of code. Because if we run the user code first and then the defaults, then we need to check again the user requests. This is because $wantceph not being set does not necessarily mean, it has not been set yet, but it could also mean, that the user requested it to be disabled. So we would need more "if"s than with the current approach.

However I agree that the variable naming is extremely misleading. I would recommend renaming the variables.
Do you prefer a variable rename within this PR (of course separate commit) or within a separate PR?